src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Ball {
+    private String color;
+    private int number;
+
+    public void setColor() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        this.color = colorSupplier.getRandomColor();
+    }
+
+    public void setNumber() {
+        Random random = new Random();
+        this.number = random.nextInt(100);
+    }
+
+    @Override
+    public String toString() {
+        return this.getClass().getSimpleName()
+                + "color: "
+                + this.color
+                + ", number: "
+                + this.number;
+    }
```

Comment:
we need `this` keyword only for when we have similar variable naming, or we want to reference a current object
```suggestion
        return this.getClass().getSimpleName()
                + " color: " + color
                + ", number: " + number;
    }
```